.circleci/config.yml

```diff
@@ -1,56 +1,171 @@
+version: 2
+
 defaults: &defaults
   working_directory: /routemaster
   docker:
-    - image: deliveroo/circleci:0.1.0
+    - image: deliveroo/circleci:0.1.8
 
-version: 2
+build_steps: &build_steps
+  steps:
+    - setup_remote_docker:
+        reusable: true
+        version: 17.05.0-ce
 
-jobs:
-  run_test_suite:
-    <<: *defaults
+    - checkout
 
-    # we want to run tests multiple times to minimise the risk of flaky tests in
-    # this (critical) application
-    parallelism: 5
+    - run:
+        name: Clean up env
+        command: ci down
+
+    - run:
+        name: Build image
+        command: |
+          docker build \
+            -f Dockerfile \
+            --build-arg PROCESS=${ROUTEMASTER_PROCESS}
+            -t "${CIRCLE_PROJECT_REPONAME}-${ROUTEMASTER_PROCESS}:${CIRCLE_SHA1}" \
+            .
+
+    - run:
+        name: Save image
+        command: |
+          mkdir -p workspace
+          docker save \
+            "${CIRCLE_PROJECT_REPONAME}-${ROUTEMASTER_PROCESS}:${CIRCLE_SHA1}" \
+            --output workspace/${CIRCLE_SHA1}.tar
+
+    - persist_to_workspace:
+        root: workspace
+        paths:
+          - '*.tar'
+
+push_steps: &push_steps
+  steps:
+  - setup_remote_docker:
+      reusable: true
+      version: 17.05.0-ce
+
+  - attach_workspace:
+      at: workspace
+
+  - run:
+      name: Import image
+      command: docker load --input workspace/${CIRCLE_SHA1}.tar
+
+  - run:
+      name: Push image to repository
+      command: |
+        `print_env ${TARGET}`
+        push_image_to_ecr \
+          --image-name "${CIRCLE_PROJECT_REPONAME}-${ROUTEMASTER_PROCESS}" \
```

Comment:
Not sure about that - why do do we need a separate image for each process?

Reply:
That's a good question. The idea was that we decide which process to run in the Dockerfile. If we can do that in Hopper, e.g. choose the command in the config file, then we don't need one image per process.

Reply:
You can pass it dynamically to the environment through command: `sh -c 'ENV=val blah'`. We could make it even easier by allowing folks to specify per-service environment inside Hopper config.